### PR TITLE
Add support for Address Map resource

### PR DESCRIPTION
**Pull request by hunts**
Adds support for the [Proxy IPs](https://developers.cloudflare.com/byoip/proxy-ips/) feature, allow CRUD address maps.

### CODE REVIEW

File: internal/sdkv2provider/provider.go [hunk 1]
```diff
@@ -357,7 +359,11 @@ func configure(version string, p *schema.Provider) func(context.Context, *schema
 		retryOpt := cloudflare.UsingRetryPolicy(int(retries), int(minBackOff), int(maxBackOff))
 		options := []cloudflare.Option{limitOpt, retryOpt, baseURL}
 
-		options = append(options, cloudflare.Debug(logging.IsDebugOrHigher()))
+		if logging.IsDebugOrHigher() {
```
Comment (jacobbednarz) [inline internal/sdkv2provider/provider.go:362]: does this need to stay? i can't say i've used it so unsure what the goal is here.

File: internal/sdkv2provider/resource_cloudflare_address_map_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccCloudflareAddressMap(t *testing.T) {
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_address_map_test.go:12]: just waiting to get this all hooked up on the acceptance testing account so these can run

File: internal/sdkv2provider/schema_cloudflare_address_map.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		},
+		"ips": {
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Description: "The set of IPs on the Address Map.",
+			Elem: &schema.Resource{
+				SchemaVersion: 1,
+				Schema: map[string]*schema.Schema{
+					"ip": {
+						Type:         schema.TypeString,
```
Comment (jacobbednarz) [inline internal/sdkv2provider/schema_cloudflare_address_map.go:49]: will want to add a `Description` here (and to the other missing fields)

File: internal/sdkv2provider/schema_cloudflare_address_map.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		},
+		"memberships": {
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Description: "Zones and Accounts which will be assigned IPs on this Address Map.",
+			Elem: &schema.Resource{
+				SchemaVersion: 1,
+				Schema: map[string]*schema.Schema{
+					"identifier": {
+						Type:     schema.TypeString,
```
Comment (jacobbednarz) [inline internal/sdkv2provider/schema_cloudflare_address_map.go:64]: same as https://github.com/cloudflare/terraform-provider-cloudflare/pull/2290/files#r1136505878

File: internal/sdkv2provider/schema_cloudflare_address_map.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+						Required: true,
+					},
+					"kind": {
+						Type:         schema.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice([]string{"account", "zone"}, false),
```
Comment (jacobbednarz) [inline internal/sdkv2provider/schema_cloudflare_address_map.go:70]: same as https://github.com/cloudflare/terraform-provider-cloudflare/pull/2290/files#r1136505878

File: docs/resources/address_map.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
+---
```
Comment (jacobbednarz) [inline docs/resources/address_map.md:1]: will re-run this after the schema descriptions are updated.

**Comment by jacobbednarz** [comment]
thanks for this @hunts! we're 95% of the way there. to summarise the inline comments:

- we need to update all fields to have a `Description` field.
- get the acceptance testing hooked up with the correct entitlements so we can run these tests.
- add a CHANGELOG entry 

after that, we should be good to go here.

File: internal/sdkv2provider/provider.go (hunk 1 above)
Comment (hunts) [reply internal/sdkv2provider/provider.go:362]: I'm not sure about the `cloudflare.Debug()` thing, what I'd like to add is the HTTP logging capability, it's helpful on figuring out what goes wrong by looking at the request and response payloads.

Comment (jacobbednarz) [reply internal/sdkv2provider/provider.go:362]: gotcha, so the existing functionality already handles this. 

[`cloudflare.Debug()`](https://github.com/cloudflare/cloudflare-go/blob/64af3d2566da4be7093bc20562c92709628c42fa/options.go#L97) logs the HTTP transaction with some [sensitive values redacted](https://github.com/cloudflare/cloudflare-go/blob/64af3d2566da4be7093bc20562c92709628c42fa/cloudflare.go#L386-L414). the tricky bit to see here is that `logging.IsDebugOrHigher()` returns a boolean based on the setting provided to `TF_LOG`. if you want debugging enabled, you use the environment variable `TF_LOG=DEBUG terraform apply ...` to see all the logs. you can also control the [formats and flow using some other environment variables](https://developer.hashicorp.com/terraform/plugin/log/managing).

Comment (jacobbednarz) [reply internal/sdkv2provider/provider.go:362]: in short, i think we can revert this and still get all the debugging goodies you are after here.

Comment (hunts) [reply internal/sdkv2provider/provider.go]: Awesome, dropped the commit. I think I tried `TF_LOG_PROVIDER=DEBUG` and it gave nothing about the HTTP payloads, thought there is no HTTP logging in the provider. Apparently, it is related to `TF_LOG`.

**Comment by jacobbednarz** [comment]
acceptance tests all passing

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareAddressMap" -timeout 120m -parallel 1
=== RUN   TestAccCloudflareAddressMap
--- PASS: TestAccCloudflareAddressMap (37.29s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	38.320s
```

**Pull request merged**